Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/FileShareServiceTest.cs

```diff
@@ -125,5 +125,40 @@ public void WhenFSSDownloadFileAsyncThrowsException_ThenShouldExecuteCatch()
             Task<Stream> result = _fileShareService.FSSDownloadFileAsync("", "", "", CorrelationId);
             Assert.IsTrue(result.IsFaulted);
         }
+
+        [Test]
+        public void WhenFSSDownloadZipAsyncFileIsCalled_ThenShouldReturnByteArray()
+        {
+            string batchId = Guid.NewGuid().ToString();
+            const string fileName = "Daily 16-05-22.zip";
+
+            IResult<Stream> stream = new Result<Stream>
+            {
+                Data = new MemoryStream(Encoding.UTF8.GetBytes("test stream"))
+            };
+
+            _fileShareServiceConfig.Value.BaseUrl = "https://filesqa.admiralty.co.uk";
+
+            A.CallTo(() => _fileShareApiClient.DownloadZipFileAsync(batchId, CancellationToken.None)).Returns(stream);
+
+            Task<Stream> result = _fileShareService.FSSDownloadZipFileAsync(batchId, fileName, FakeAccessToken, CorrelationId);
+
+            Assert.IsInstanceOf<Task<Stream>>(result);
+        }
+
+        [Test]
+        public void WhenFSSDownloadZipFileAsyncThrowsException_ThenShouldExecuteCatch()
+        {
+            string batchId = Guid.NewGuid().ToString();
+            const string fileName = "Daily 16-05-22.zip";
+
+            _fileShareServiceConfig.Value.BaseUrl = null;
+
+            A.CallTo(() => _fileShareApiClient.DownloadZipFileAsync(A<string>.Ignored, CancellationToken.None)).Throws(new Exception());
+
+            Task<Stream> result = _fileShareService.FSSDownloadZipFileAsync(batchId, fileName, FakeAccessToken, CorrelationId);
```

Comment:
the fake AccessToken i take it is due to the fact we are not testing in the QA environment. same as above though if thats the case as we will need to constantly update the code depending on the environment its running against. overhead

Reply:
We have changed the base URL to dummy url as  : "_https://www.test.com/_"

Reply:
how is the token implemented now please can you explain. thank you

Reply:
Now we have added FSS client as a dependency so instead of referring it directly we are able to mock it